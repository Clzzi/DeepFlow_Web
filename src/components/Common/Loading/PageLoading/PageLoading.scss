.PageLoading {
  width: 100%;
  height: 80vh;
  display: flex;
  display: -webkit-flex;
  flex-flow: column wrap;
  justify-content: center;
  align-items: center;
  font-size: var(--font-contents);
  color: var(--contrast);
}
