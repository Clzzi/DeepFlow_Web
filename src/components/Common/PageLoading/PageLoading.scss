.PageLoading {
	width: 100%;
	height: 80vh;
	display: flex;
	display: -webkit-flex;
	flex-flow: column wrap;
	justify-content: center;
	align-items: center;
	font-size: 1.5rem;
}
