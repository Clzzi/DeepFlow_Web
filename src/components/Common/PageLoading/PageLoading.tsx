import classNames from 'classnames';
import { ClassNamesFn } from 'classnames/types';
import { ScaleLoader } from 'react-spinners';
import { palette } from 'styles/Palette/Palette';

const style = require('./PageLoading.scss');
const cx: ClassNamesFn = classNames.bind(style);

interface PageLoadingProps {
  text: string;
}

const PageLoading = ({ text }: PageLoadingProps) => {
  return (
    <div className={cx('PageLoading')}>
      <ScaleLoader color={palette.main} />
      <div>{text}</div>
    </div>
  );
};

export default PageLoading;
