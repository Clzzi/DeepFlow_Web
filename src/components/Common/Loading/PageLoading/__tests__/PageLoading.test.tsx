import { render } from '@testing-library/react';
import PageLoading, { PageLoadingProps } from '../PageLoading';

describe('PageLoading', () => {
  const initialSetup = (props: Partial<PageLoadingProps> = {}) => {
    const initialProps: PageLoadingProps = {
      text: '로딩중',
      ...props,
    };

    return render(<PageLoading {...initialProps} />);
  };

  it('render correctly', () => {
    const { container } = initialSetup();
    expect(container).toMatchSnapshot();
  });
});