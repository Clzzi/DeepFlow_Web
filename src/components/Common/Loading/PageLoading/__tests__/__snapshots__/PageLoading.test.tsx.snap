// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLoading render correctly 1`] = `
<div>
  <div
    class="PageLoading"
  >
    <span
      class="css-0"
    >
      <span
        class="css-x63co1"
      />
      <span
        class="css-89r7oh"
      />
      <span
        class="css-3cy2mi"
      />
      <span
        class="css-n3yql8"
      />
      <span
        class="css-1rxpohb"
      />
    </span>
    <div>
      로딩중
    </div>
  </div>
</div>
`;
